internal/provider/resource_open_acess.go

```diff
@@ -0,0 +1,259 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/booldefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ resource.Resource                = &OpenAccessResource{}
+	_ resource.ResourceWithConfigure   = &OpenAccessResource{}
+	_ resource.ResourceWithImportState = &OpenAccessResource{}
+)
+
+func SquaredUpOpenAccessResource() resource.Resource {
+	return &OpenAccessResource{}
+}
+
+type OpenAccessResource struct {
+	client *SquaredUpClient
+}
+
+type DashboardSharing struct {
+	OpenAccessID          types.String `tfsdk:"id"`
+	DashboardID           types.String `tfsdk:"dashboard_id"`
+	WorkspaceID           types.String `tfsdk:"workspace_id"`
+	RequireAuthentication types.Bool   `tfsdk:"require_authentication"`
+	EnableLink            types.Bool   `tfsdk:"enable_link"`
+	OpenAccessLink        types.String `tfsdk:"open_access_link"`
+	LastUpdated           types.String `tfsdk:"last_updated"`
+}
+
+func (r *OpenAccessResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dashboard_share"
+}
+
+func (r *OpenAccessResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Enable Open Access for a dashboard",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of the Open Access",
```

Comment:
The ID of the Open Access dashboard

Reply:
Based on the comment below I have changed it to The ID of the Shared dashboard